kratos/includes/node.h

```diff
@@ -835,7 +826,7 @@ class Node : public Point,  public IndexedObject, public Flags
             return *(it_dof.base());
         }
 
-        typename DofType::Pointer p_new_dof =  Kratos::make_shared<DofType>(Id(), &mSolutionStepsNodalData, rDofVariable);
+        typename DofType::Pointer p_new_dof =  Kratos::make_shared<DofType>(&mNodalData, rDofVariable);

```

Comment:
already discussed, but i put this as a reminder:
dofs have a unique owner (the node) so it does not make sense to use a shred pointer to hold them.

This is best done incrementally within another PR, so in a sense this is noise here, nevertheless it will allow us to get rid of 2 more doubles per dof so it is relevant to recall it here

Reply:
I think there is already an issue abt this => #2847